Autopilot/Src/MPXV7002DP.cpp

```diff
@@ -0,0 +1,63 @@
+#include "airspeed.hpp"
+#include "adc.h"
+#include <stddef.h>
+#include <cmath>
+
+#define VS 5
+#define DENSITY_AIR 1.2754
+#define ADC_RESOLUTION 4096
+
+
+MPXV7002DP::MPXV7002DP() {
+    //constructor
+    MX_ADC3_Init();
+    HAL_ADC_START(&hadc3);
+    //start the adc or whatever
+
+}
+
+MPXV7002DP* MPXV7002DP::s_Instance = NULL;
+
+MPXV7002DP* MPXV7002DP::GetInstance() {
+    if(!s_Instance) {
+        s_Instance = new MPXV7002DP;
+    }
+    return s_Instance;
+}
+
+void MPXV7002DP::GetResult(airspeedData_t *Data) {
+        Data->airspeed = airspeed;
+        Data->isDataNew = dataNew;
+        dataNew = false;
+        Data->sensorStatus = sensorStatus;
+        Data->utcTime = timeOfLastResult;
+}
+
+void MPXV7002DP::Begin_Measuring() {
```

Comment:
The Hal_adc_getValue function litterally just reads a register. I.E. It's completely non blocking.
![image](https://user-images.githubusercontent.com/32182266/99277400-c4fb2500-27fb-11eb-92f7-1b4a1002fca6.png)
So there's no need to give this sensor a thread to begin the measurement process. You should give begin_measuring a blank implementation and explain that it doesn't need to be called in the header, then shift all the code in begin_measuring to get_result. Bonus is that data will be guaranteed to be new on every call.